odl/solvers/advanced/proximal_operators.py

```diff
@@ -231,18 +409,68 @@ def proximal_convexconjugate_l2(space, lam=1, g=None):
 
     Returns
     -------
-    prox : `Operator`
-        Returns the proximal operator to be initialized
+    prox : `callable`
+        Factory for the proximal operator to be initialized
+
+    Notes
+    -----
+    Most problems are forumlated for the norm squared, in that case use that
+    proximal operator instead.
+
+    See Also
+    --------
+    proximal_l2 : proximal without convex conjugate
+    proximal_convexconjugate_l2 : proximal without square
     """
+
+    prox_l2 = proximal_l2(space, lam=lam, g=g)
+    return proximal_convexconjugate(prox_l2)
+
+
+def proximal_l2(space, lam=1, g=None):
+    """Proximal operator factory of the l2-norm.
+
+    Function for the proximal operator of the convex conjugate of the
+    functional F where F is the l2-norm
+
+        F(x) =  lam ||x - g||_2
+
+    The proximal operator of ``sigma * F`` where ``sigma`` is a step size
+    is given by
+
+        prox[sigma * F^*](y) = (y - sigma g) / (1 + sigma/lam)
+
+    Parameters
+    ----------
+    space : `DiscreteLp` or `ProductSpace` of `DiscreteLp`
+        Domain of F(x)
+    g : `DiscreteLpVector`
+        An element in ``space``
+    lam : positive `float`
+        Scaling factor or regularization parameter
+
+    Returns
+    -------
+    prox : `callable`
+        Factory for the proximal operator to be initialized
+
+    Notes
+    -----
+    Most problems are forumlated for the norm squared, in that case use that
+    proximal operator instead.
+
+    See Also
+    --------
+    proximal_l2_squared : proximal for norm squared
+    proximal_convexconjugate_l2 : proximal for convex conjugate
+    """
+
     lam = float(lam)
 
-    if g is None:
-        g = space.zero()
-    else:
-        if g not in space:
-            raise TypeError('{!r} is not an element of {!r}'.format(g, space))
+    if g is not None and g not in space:
+        raise TypeError('{!r} is not an element of {!r}'.format(g, space))
 
-    class _ProximalConvConjL2(Operator):
+    class _ProximalL2(Operator):
```

Comment:
Why exactly do we use underscores for the returned operators? That makes them print themselves in an ugly way.